File: app/android/app/src/main/kotlin/com/mednote/models/Visit.kt
package com.mednote.models

import kotlinx.serialization.Serializable

@Serializable
data class Visit(
    val id: String = "v${System.currentTimeMillis()}",
    val doctor: String = "Phone Visit",
    val specialty: String = "",
    val date: String = VisitFormats.formatDate(),
    val time: String = VisitFormats.formatTime(),
    val duration: String = "0m 0s",
    val originalLang: String = "en",
    val originalTranscript: List<String> = emptyList(),
    val translatedTranscript: List<String>? = null,
    val summary: String? = null,
    val translatedSummary: String? = null,
    val audioFileName: String? = null,
    val extractedItems: List<ExtractedItem> = emptyList(),
    val isProcessing: Boolean = false
)

@Serializable
data class ExtractedItem(
    val id: String,
    val type: String,
    val text: String,
    val time: String = "",
    val suggestedHour: Int = 8,
    val status: String = "provisional"
)

enum class AppLanguage(val code: String, val label: String) {
    EN("en", "English"),
    ZH("zh", "中文"),
    FR("fr", "Français"),
    DE("de", "Deutsch"),
    JA("ja", "日本語"),
    KO("ko", "한국어"),
    ES("es", "Español"),
    HI("hi", "हिंदी"),
    VI("vi", "Tiếng Việt"),
    AR("ar", "العربية");

    companion object {
        fun fromCode(code: String): AppLanguage =
            entries.find { it.code == code } ?: EN
    }
}

enum class ProcessingStep(val label: String) {
    SAVING("Saving recording…"),
    TRANSCRIBING("Transcribing audio…"),
    ANALYZING("Creating summary…"),
    DONE("Done!")
}

enum class CallVisitPhase {
    IDLE, RECORDING, ON_CALL, PROCESSING
}

object VisitFormats {
    fun formatDate(): String {
        val c = java.util.Calendar.getInstance()
        val months = arrayOf("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")
        return "${months[c.get(java.util.Calendar.MONTH)]} ${c.get(java.util.Calendar.DAY_OF_MONTH)}, ${c.get(java.util.Calendar.YEAR)}"
    }

    fun formatTime(): String {
        val sdf = java.text.SimpleDateFormat("h:mm a", java.util.Locale.US)
        return sdf.format(java.util.Date())
    }

    fun formatDuration(seconds: Int): String {
        if (seconds <= 0) return "uploaded"
        return "${seconds / 60}m ${seconds % 60}s"
    }
}

File: app/ios/MedNote/Models/Visit.swift
import Foundation

struct Visit: Identifiable, Codable, Equatable, Hashable {
    var id: String
    var doctor: String
    var specialty: String
    var date: String
    var time: String
    var duration: String
    var originalLang: String
    var originalTranscript: [String]
    var translatedTranscript: [String]?
    var summary: String?
    var translatedSummary: String?
    var audioFileName: String?
    var extractedItems: [ExtractedItem]
    var isProcessing: Bool

    init(
        id: String = "v\(Int(Date().timeIntervalSince1970 * 1000))",
        doctor: String = "Phone Visit",
        specialty: String = "",
        date: String = Visit.formatDate(Date()),
        time: String = Visit.formatTime(Date()),
        duration: String = "0m 0s",
        originalLang: String = "en",
        originalTranscript: [String] = [],
        translatedTranscript: [String]? = nil,
        summary: String? = nil,
        translatedSummary: String? = nil,
        audioFileName: String? = nil,
        extractedItems: [ExtractedItem] = [],
        isProcessing: Bool = false
    ) {
        self.id = id
        self.doctor = doctor
        self.specialty = specialty
        self.date = date
        self.time = time
        self.duration = duration
        self.originalLang = originalLang
        self.originalTranscript = originalTranscript
        self.translatedTranscript = translatedTranscript
        self.summary = summary
        self.translatedSummary = translatedSummary
        self.audioFileName = audioFileName
        self.extractedItems = extractedItems
        self.isProcessing = isProcessing
    }

    static func formatDate(_ date: Date) -> String {
        let f = DateFormatter()
        f.dateFormat = "MMM d, yyyy"
        f.locale = Locale(identifier: "en_US")
        return f.string(from: date)
    }

    static func formatTime(_ date: Date) -> String {
        let f = DateFormatter()
        f.dateStyle = .none
        f.timeStyle = .short
        return f.string(from: date)
    }

    static func formatDuration(seconds: Int) -> String {
        guard seconds > 0 else { return "uploaded" }
        return "\(seconds / 60)m \(seconds % 60)s"
    }
}

struct ExtractedItem: Identifiable, Codable, Equatable {
    var id: String
    var type: ItemType
    var text: String
    var time: String
    var suggestedHour: Int
    var status: String

    enum ItemType: String, Codable {
        case medication
        case appointment
        case action
    }
}

enum AppLanguage: String, CaseIterable, Identifiable {
    case en, zh, fr, de, ja, ko, es, hi, vi, ar

    var id: String { rawValue }

    var label: String {
        switch self {
        case .en: return "English"
        case .zh: return "中文"
        case .fr: return "Français"
        case .de: return "Deutsch"
        case .ja: return "日本語"
        case .ko: return "한국어"
        case .es: return "Español"
        case .hi: return "हिंदी"
        case .vi: return "Tiếng Việt"
        case .ar: return "العربية"
        }
    }

    var whisperCode: String? {
        rawValue
    }
}

enum ProcessingStep: Int, CaseIterable {
    case saving = 1
    case transcribing
    case analyzing
    case done

    var label: String {
        switch self {
        case .saving: return "Saving recording…"
        case .transcribing: return "Transcribing audio…"
        case .analyzing: return "Creating summary…"
        case .done: return "Done!"
        }
    }
}

enum CallVisitPhase: Equatable {
    case idle
    case recording
    case onCall
    case processing
}
